rclcpp/include/rclcpp/client.hpp

```diff
@@ -119,7 +120,7 @@ namespace node_interfaces
 class NodeBaseInterface;
 }  // namespace node_interfaces
 
-class ClientBase
+class ClientBase: public sched::SchedBase
```

Comment:
What exactly does using `SchedBase` buy you here? From what I can see, it effectively just adds a `SchedAttr` member to each of these callback class types (ClientBase, TimerBase, SubscriptionBase, ServiceBase). How would that be different from just adding a SchedAttr member variable to each of these 4 classes directly?

I thought the real issue is that `Executor::execute_any_executable` in execuctor.cpp is still a _single_ function which checks the type of AnyExecutable you've given it, and then executes the correct `execute` function for that callback type. I thought we would do something like make a virtual `execute` function in `SchedBase`, and then implement each callback type's specific `execute` function within its representative derived class.

Reply:
Rejected. Discussed offline. The ROS implementation has no timing/functional issue. The proposal only improves from the OOP perspective. This is orthogonal to this PR, a separate PR should be created to refactor ROS.